tests/feedback.test.ts

```diff
@@ -0,0 +1,195 @@
+    const [feedback] = FeedbackFactory.create(1);
+
+    const feedbackController = ControllerFactory.feedback(conn);
+    await new PortalState().createUsers([user]).write(conn);
+
+    await ControllerFactory.feedback(conn).submitFeedback({ feedback }, user);
+    const submittedFeedbackResponse = await feedbackController.getFeedback(user);
+
+    expect(submittedFeedbackResponse.feedback).toHaveLength(1);
+    expect(submittedFeedbackResponse.feedback[0]).toStrictEqual({
+      ...submittedFeedbackResponse.feedback[0],
+      user: user.getPublicProfile(),
+      status: FeedbackStatus.SUBMITTED,
+      ...feedback,
+    });
+  });
+
+  test('is invalidated when submission description is too short', async () => {
+    const [feedback] = FeedbackFactory.with({ description: 'A short description' });
+
+    const errors = await validate(plainToClass(Feedback, feedback));
+
+    expect(errors).toBeDefined();
+    expect(errors).toHaveLength(1);
+    expect(errors[0].property).toEqual('description');
+    expect(errors[0].constraints.minLength).toBeDefined();
+  });
+
+  test('persists submission data and activity in database on successful submission', async () => {
+    const conn = await DatabaseConnection.get();
+    const [user] = UserFactory.create(1);
+    const [feedback] = FeedbackFactory.create(1);
+
+    await new PortalState().createUsers([user]).write(conn);
+
+    const submittedFeedbackResponse = await ControllerFactory.feedback(conn).submitFeedback({ feedback }, user);
+    const [persistedFeedback] = await conn.manager.find(FeedbackModel, { relations: ['user'] });
+    const activity = await conn.manager.find(ActivityModel, { relations: ['user'] });
+    const feedbackSubmissionActivity = { ...activity[1] };
+
+    expect(submittedFeedbackResponse.feedback).toStrictEqual(persistedFeedback.getPublicFeedback());
+    expect(feedbackSubmissionActivity).toStrictEqual({
+      ...feedbackSubmissionActivity,
+      type: ActivityType.SUBMIT_FEEDBACK,
+      scope: ActivityScope.PRIVATE,
+      user,
+    });
+  });
+
+  test('admins can view feedback from any user', async () => {
+    const conn = await DatabaseConnection.get();
+    const [user1, user2] = UserFactory.create(2);
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const [feedback1, feedback2] = FeedbackFactory.create(2);
+
+    const feedbackController = ControllerFactory.feedback(conn);
+    await new PortalState().createUsers([user1, user2, admin]).write(conn);
+
+    const submittedFeedback1Response = await feedbackController.submitFeedback({ feedback: feedback1 }, user1);
+    const submittedFeedback2Response = await feedbackController.submitFeedback({ feedback: feedback2 }, user2);
+    const allSubmittedFeedbackResponse = await feedbackController.getFeedback(admin);
+
+    expect(submittedFeedback1Response.feedback).toMatchObject(feedback1);
+    expect(submittedFeedback2Response.feedback).toMatchObject(feedback2);
```

Comment:
I still don't understand the need to verify the submission responses.

Reply:
Yeah you're right, this assertion is pretty redundant here